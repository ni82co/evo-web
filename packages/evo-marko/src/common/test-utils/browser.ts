import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el: HTMLElement, to: number, cb: Function) {
  fireEvent.scroll(el);
  el.scrollLeft = to;
  setTimeout(cb, 600);
}

function waitFrames(count: number, cb: Function) {
  if (count) {
    return requestAnimationFrame(() => {
      waitFrames(count - 1, cb);
    });
  }

  cb();
}

interface FastAnimations {
  fastAnimationStyle?: HTMLStyleElement;
  start: () => void;
  stop: () => void;
}

const fastAnimations:FastAnimations = {
  // Adds an style to the document which forces all transitions to run more quickly for the tests.
  start() {
    if (this.fastAnimationStyle) {
      return;
    }

    this.fastAnimationStyle = document.createElement("style");
    this.fastAnimationStyle.innerHTML = `* {
                transition-duration: 0.1s !important;
                transition-delay: 0.1s !important;
                scroll-behavior: auto !important;
            }`;
    document.head.appendChild(this.fastAnimationStyle);
  },
  stop() {
    if (!this.fastAnimationStyle) {
      return;
    }

    document.head.removeChild(this.fastAnimationStyle);
    this.fastAnimationStyle = undefined;
  },
};

export { fastAnimations, waitFrames, simulateScroll };
