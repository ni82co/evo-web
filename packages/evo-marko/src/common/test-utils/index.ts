export * from "./browser"
export * from "./snapshots";