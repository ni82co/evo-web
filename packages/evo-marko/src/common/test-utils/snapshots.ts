import { render, normalize } from "@marko/testing-library";
import { expect } from "vitest";

async function snapshotHTML(template: any, input?: any) {
  const { container } = await render(template, input);
  expect(normalize(container)).toMatchSnapshot();
}

export { snapshotHTML };
