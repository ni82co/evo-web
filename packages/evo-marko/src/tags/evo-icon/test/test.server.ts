import { describe, it } from "vitest";

import { snapshotHTML } from "../../../common/test-utils/snapshots";
import template from "../index.marko";

const iconName = "mic";
const progrmaBadgeIcon = "program-badge-authenticity-guaranteed";

describe("icon", () => {
  it("renders icon inline type with title text", async () => {
    const input = {
      _name: iconName,
      _type: "icon",
      _symbol: () => {},
      a11yText: "inline icon",
    };
    await snapshotHTML(template, input);
  });

  it("renders icon inline type without title text", async () => {
    const input = {
      _name: iconName,
      _type: "icon",
      _symbol: () => {}
    };
    await snapshotHTML(template, input);
  });

  it("renders icon no-skin-classes", async () => {
    const input = {
      _name: iconName,
      _type: "icon",
      class: "custom-class",
      _symbol: () => {},
    };
    await snapshotHTML(template, input);
  });
});
