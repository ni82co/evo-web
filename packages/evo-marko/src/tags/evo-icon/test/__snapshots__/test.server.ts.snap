// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`icon > renders icon inline type with title text 1`] = `
<DocumentFragment>
  <svg
    aria-labelledby="GENERATED-0"
    class="icon icon--mic"
    role="img"
  >
    <use
      href="#icon-mic"
    />
    <title
      id="GENERATED-0"
    >
      inline icon
    </title>
  </svg>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.a=[0,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",1];M._.w()
  </script>
</DocumentFragment>
`;

exports[`icon > renders icon inline type without title text 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="icon icon--mic"
  >
    <use
      href="#icon-mic"
    />
  </svg>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.a=[0,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",1];M._.w()
  </script>
</DocumentFragment>
`;

exports[`icon > renders icon no-skin-classes 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="custom-class icon icon--mic"
  >
    <use
      href="#icon-mic"
    />
  </svg>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.a=[0,{htmlInput:{}}]),"src/tags/evo-icon/index.marko_0_htmlInput",1];M._.w()
  </script>
</DocumentFragment>
`;
